contracts/contracts/TallyNonQv.sol

```diff
@@ -0,0 +1,275 @@
+  /// @param _proof the proof generated after tallying this batch
+  function tallyVotes(uint256 _newTallyCommitment, uint256[8] calldata _proof) public onlyOwner {
+    _votingPeriodOver(poll);
+    updateSbCommitment();
+
+    // get the batch size and start index
+    (uint8 intStateTreeDepth, , , ) = poll.treeDepths();
+    uint256 tallyBatchSize = TREE_ARITY ** intStateTreeDepth;
+    uint256 batchStartIndex = tallyBatchNum * tallyBatchSize;
+
+    // save some gas because we won't overflow uint256
+    unchecked {
+      tallyBatchNum++;
+    }
+
+    (uint256 numSignUps, ) = poll.numSignUpsAndMessages();
+
+    // Require that there are untallied ballots left
+    if (batchStartIndex >= numSignUps) {
+      revert AllBallotsTallied();
+    }
+
+    bool isValid = verifyTallyProof(_proof, numSignUps, batchStartIndex, tallyBatchSize, _newTallyCommitment);
+
+    if (!isValid) {
+      revert InvalidTallyVotesProof();
+    }
+
+    // Update the tally commitment and the tally batch num
+    tallyCommitment = _newTallyCommitment;
+  }
+
+  /// @notice Verify the tally proof using the verifying key
+  /// @param _proof the proof generated after processing all messages
+  /// @param _numSignUps number of signups for a given poll
+  /// @param _batchStartIndex the number of batches multiplied by the size of the batch
+  /// @param _tallyBatchSize batch size for the tally
+  /// @param _newTallyCommitment the tally commitment to be verified at a given batch index
+  /// @return isValid whether the proof is valid
+  function verifyTallyProof(
+    uint256[8] calldata _proof,
+    uint256 _numSignUps,
+    uint256 _batchStartIndex,
+    uint256 _tallyBatchSize,
+    uint256 _newTallyCommitment
+  ) public view returns (bool isValid) {
+    (uint8 intStateTreeDepth, , , uint8 voteOptionTreeDepth) = poll.treeDepths();
+
+    (IMACI maci, , ) = poll.extContracts();
+
+    // Get the verifying key
+    VerifyingKey memory vk = vkRegistry.getTallyVk(maci.stateTreeDepth(), intStateTreeDepth, voteOptionTreeDepth);
+
+    // Get the public inputs
+    uint256 publicInputHash = genTallyVotesPublicInputHash(
+      _numSignUps,
+      _batchStartIndex,
+      _tallyBatchSize,
+      _newTallyCommitment
+    );
+
+    // Verify the proof
+    isValid = verifier.verify(_proof, vk, publicInputHash);
+  }
+
+  /// @notice Compute the merkle root from the path elements
+  /// and a leaf
+  /// @param _depth the depth of the merkle tree
+  /// @param _index the index of the leaf
+  /// @param _leaf the leaf
+  /// @param _pathElements the path elements to reconstruct the merkle root
+  /// @return current The merkle root
+  function computeMerkleRootFromPath(
+    uint8 _depth,
+    uint256 _index,
+    uint256 _leaf,
+    uint256[][] calldata _pathElements
+  ) internal pure returns (uint256 current) {
+    uint256 pos = _index % TREE_ARITY;
+    current = _leaf;
+    uint8 k;
+
+    uint256[TREE_ARITY] memory level;
```

Comment:
Let's initialize it.

Reply:
initialize to what?

Reply:
here the warning to get more context https://github.com/privacy-scaling-explorations/maci/security/code-scanning/534

Reply:
mm that's odd to me, I always knew that variables are init automatically to their default value, and doing it manually is unnecessary.. not sure why this is complaining here 🤔